### PR TITLE
android: Remove usage of AHB_VALIDATION_SUPPORT

### PR DESCRIPTION
**Pull request by juan-lunarg**
Removes usage of AHB_VALIDATION_SUPPORT and cleans up surrounding Android code.

Use C++17 std::from_chars to convert string to integer. It's also much safer to use considering it's non-allocating and non-throwing.

### CODE REVIEW

File: layers/vk_layer_config.cpp [hunk 1]
```diff
@@ -413,23 +413,32 @@ void PrintMessageType(VkFlags vk_flags, char *msg_flags) {
...
+#include <charconv>
+
 // This catches before dlopen fails if the default Android-26 layers are being used and attempted to be ran on Android 25 or below
-#if defined(__ANDROID__)
-#include "utils/android_ndk_types.h"  // get AHB_VALIDATION_SUPPORT macro
-void __attribute__((constructor)) CheckAndroidVersion();
-void CheckAndroidVersion() {
-#ifdef AHB_VALIDATION_SUPPORT
-    string version_env = GetEnvironment("ro.build.version.sdk");
-    int target_version = atoi(version_env.c_str());
```
Comment (juan-lunarg) [inline layers/vk_layer_config.cpp:423]: See issues with using atoi (https://releases.llvm.org/16.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/cert/err34-c.html).

Which is why I decided to use C++17 std::from_chars

https://en.cppreference.com/w/cpp/utility/from_chars

File: layers/core_checks/cc_android.cpp [hunk 2]
```diff
@@ -21,7 +21,7 @@
 #include "generated/vk_enum_string_helper.h"
 #include "core_validation.h"
 
-#ifdef AHB_VALIDATION_SUPPORT
```
Comment (juan-lunarg) [inline layers/core_checks/cc_android.cpp:24]: AHB_VALIDATION_SUPPORT isn't needed as of https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/5864

File: layers/vk_layer_config.cpp [hunk 3]
```diff
@@ -413,23 +413,32 @@ void PrintMessageType(VkFlags vk_flags, char *msg_flags) {
...
+    uint32_t target_version{};
+
+    if (std::from_chars(version.data(), version.data() + version.size(), target_version).ec != std::errc()) {
+        return;
+    }
+
+    if (target_version < 26) {
         LOGCONSOLE(
             "ERROR - Targeted Android version is %d and needs to be 26 or above. Please read "
             "https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/main/BUILD.md for how to build the Validation Layers "
```
Comment (spencer-lunarg) [inline layers/vk_layer_config.cpp:438]: So this is not a valid thing now right? We are removing support for 24/25 with the removal of AHB_VALIDATION_SUPPORT

**Review by spencer-lunarg — APPROVED** [review]
LGTM - Just want to make sure it is documented well that we did support 24/25 but now aren't because of <insert reasons>

**Review by ncesario-lunarg — APPROVED** [review]
LGTM once Spencer's comment is resolved.

File: tests/CMakeLists.txt [hunk 4]
```diff
@@ -151,8 +151,6 @@ target_compile_definitions(vk_layer_validation_tests PRIVATE VVL_TESTS_USE_CMAKE
 
 add_dependencies(vk_layer_validation_tests vvl)
 
-target_include_directories(vk_layer_validation_tests PRIVATE .)
-
```
Comment (juan-lunarg) [inline tests/CMakeLists.txt:155]: This was the main source of my CI troubles. The NDK build doesn't have this include directory which I was relying on. So I'm removing it for compatibility.

**Pull request merged**